alphagov/paas-release-ci
# [#155037639] Enable zendesk in paas-product-page

https://www.pivotaltracker.com/story/show/155037639

What?
----

❗️ This PR contains a temporary commit which should be removed before merge.

We want to use zendesk instead of deskpro in the product page, so we did the PR https://github.com/alphagov/paas-product-page/pull/59

In this PR we enable that feature by:

 - Adding the credentials
 - Enabling the feature.

We also refactor how the pipeline for deploying paas-product-page is configured, trying to simplify the process and maintenance. See the commits for details.

Dependencies and how to merge
------------

 - Merge and generate the container for https://github.com/alphagov/paas-docker-cloudfoundry-tools/pull/124
 - Update the corresponding commit in this branch
 - Review first: https://github.com/alphagov/paas-product-page/pull/59 **but do not merge**
 - Pause the pipeline https://concourse.build.ci.cloudpipeline.digital/teams/main/pipelines/paas-product-page
 - Merge this PR, update the pipeline
 - Merge: https://github.com/alphagov/paas-product-page/pull/59 and unpause the pipeline

How to review?
--------------

You can try this pipeline in your dev environment by running:

```
export DEPLOY_ENV=....
make dev plain-pipelines \
    CONCOURSE_URL=https://deployer.hector.dev.cloudpipeline.digital \
    CF_USER=admin \
    CF_PASSWORD=...

```

Optionally, to check that the `setup` still works, you can deploy the `setup` pipeline in your `deployer` by:
  1. Disable terraform in the file `pipelines/setup.yml` (add a `echo`)
  2. Run it with: `make dev DEPLOY_ENV=... pipelines CONCOURSE_URL=https://deployer.${DEPLOY_ENV}.dev.cloudpipeline.digital CF_USER=admin CF_PASSWORD=... BRANCH=$(git rev-parse --abbrev-ref HEAD)`


Who?
----

Anyone but @keymon
